esi_leap/db/sqlalchemy/api.py

```diff
@@ -133,23 +133,138 @@ def offer_get_all(filters):
     start = filters.pop('start_time', None)
     end = filters.pop('end_time', None)
 
+    a_start = filters.pop('available_start_time', None)
+    a_end = filters.pop('available_end_time', None)
+
+    if a_start and a_end and a_end <= a_start:
+        raise exception.InvalidAvailabilityCommand(a_start=a_start,
+                                                   a_end=a_end)
+
     query = query.filter_by(**filters)
 
     if start and end:
         query = query.filter((start >= models.Offer.start_time) &
                              (end <= models.Offer.end_time))
 
+    for o in query:
```

Comment:
This only needs to be done if a_start and a_end are defined, right? Otherwise it shouldn't be run at all?

Reply:
youre correct. I added a check.